google/api_versions.go

```diff
@@ -42,22 +48,30 @@ type TerraformResourceData interface {
 
 // Compare the fields set in schema against a list of features and their versions to determine
 // what version of the API is required in order to manage the resource.
-func getComputeApiVersion(d TerraformResourceData, resourceVersion ComputeApiVersion, features []Feature) ComputeApiVersion {
-	versions := map[ComputeApiVersion]struct{}{resourceVersion: struct{}{}}
+func getApiVersion(d TerraformResourceData, resourceVersion ApiVersion, features []Feature, maxVersionFunc func(map[ApiVersion]struct{}) ApiVersion) ApiVersion {
+	versions := map[ApiVersion]struct{}{resourceVersion: struct{}{}}
 	for _, feature := range features {
 		if feature.InUseByDefault(d) {
 			versions[feature.Version] = struct{}{}
 		}
 	}
 
-	return maxVersion(versions)
+	return maxVersionFunc(versions)
+}
+
+func getComputeApiVersion(d TerraformResourceData, resourceVersion ApiVersion, features []Feature) ApiVersion {
```

Comment:
Why using two separate methods for compute and container? Do you envision any per product family customization happening here?

Reply:
It's more just to hide implementation details. The caller doesn't really need to know that the way we figure out the version to use is via a function that finds the max version used.

Reply:
I missed the `maxComputeVersion` and `maxContainerVersion`. I thought it was only a pass-through. All good then :)